server/integration/infinispan/src/main/java/org/infinispan/server/infinispan/task/ServerTaskRegistryImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.infinispan.server.infinispan.task;
+
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.tasks.ServerTask;
+import org.infinispan.tasks.Task;
+import org.infinispan.tasks.TaskManager;
+import org.infinispan.tasks.impl.TaskManagerImpl;
+
+import java.util.*;
+
+/**
+ * Author: Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * Date: 1/20/16
+ * Time: 12:53 PM
+ */
+@Scope(Scopes.GLOBAL)
+public class ServerTaskRegistryImpl implements ServerTaskRegistry {
+
+   @Inject
+   public void init(TaskManager taskManager, EmbeddedCacheManager cacheManager) {
+      ServerTaskEngine engine = new ServerTaskEngine(this, cacheManager);
+      ((TaskManagerImpl) taskManager).registerTaskEngine(engine);
+   }
+
+   private Map<String, ServerTaskWrapper> tasks = new HashMap<>();
```

Comment:
What will happen if we add (and get) multiple tasks at the same time? This might happen when will have multiple deployments.